src/renderer/viz/expressions/ramp.js

```diff
@@ -120,10 +122,24 @@ export default class Ramp extends BaseExpression {
     }
 
     eval (feature) {
-        this.palette = this._calcPaletteValues(this.palette);
+        const input = this.input.eval(feature);
+        if (this.palette.type === 'number-array') {
+            const m = (input - this.minKey) / (this.maxKey - this.minKey);
```

Comment:
IMO it would be great to extract this if-case to its own function, to improve legibility, maybe something like `_evalNumberArrayWithMix` or a better name

Reply:
:+1: